File: atlaskv-java-sdk/src/main/java/com/atlaskv/sdk/models/KeyValue.java
package com.atlaskv.sdk.models;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Immutable representation of a Key-Value pair and its associated version metadata.
 *
 * @param key       the unique key
 * @param value     the stored value (null if not exists)
 * @param exists    whether the key exists in the store
 * @param version   the version number of the key
 * @param createdAt epoch timestamp when the key was created
 * @param updatedAt epoch timestamp when the key was last updated
 */
public record KeyValue(
        String key,
        String value,
        @JsonProperty("found")
        @JsonAlias("exists")
        boolean exists,
        Long version,
        Long createdAt,
        Long updatedAt
) {}

File: atlaskv-ts-sdk/src/models/KeyValue.ts
/**
 * Represents the details and metadata of a key-value entry in AtlasKV.
 */
export interface KeyValue {
  /** The unique key identifier. */
  key: string;
  /** The value associated with the key (null if the key does not exist, is deleted, or expired). */
  value: string | null;
  /** Indicates whether the key currently exists in the store. */
  exists: boolean;
  /** The modification version of the key (null if metadata is excluded or the key does not exist). */
  version: number | null;
  /** Epoch timestamp in milliseconds indicating when the key was created. */
  createdAt: number | null;
  /** Epoch timestamp in milliseconds indicating when the key was last updated. */
  updatedAt: number | null;
}
